### ISSUE TITLE
i am willing to update

### ISSUE THREAD

**Issue by Queenhelena** [Xcelled/mabicommerce#24]
im trying to update it for irias stuff but dont know where you got some of the data so maybe you be down to share your discord and we can chat some time

**Comment by Xcelled** [comment]
Which data are you unsure where it came from? The data should all be in the XML from the client's pack files except for the waypoint info which I collected manually.

**Comment by Queenhelena** [comment]
so are the weights you used for the posts data are they in the clients pack files i havent looked at the client files
and how did you collect the way points manually?

**Comment by Xcelled** [comment]
Yes, the weights are in the client XML files. It's been 7 years or so since I last did anything with Mabinogi, but I believe the xml file was called commerce.xml?

To collect the waypoints, you'll need a way to view the coordinates of your character. I believe I just used [MabiPale](https://github.com/exectails/MabiPale2) to check on the coordinates of the Move packets my client sent when I clicked where I wanted a waypoint to be. I chose waypoints mostly by gut feel - basically you'll want to try to choose waypoints so they form a series of straight lines that's nearly as short a route as possible without going through unwalkable areas.

You can use the [WaypointViewer](https://github.com/Xcelled/mabicommerce/tree/master/WaypointViewer) as debug tool for this - You should end up with something that looks like this (taken from the [readme](https://github.com/Xcelled/mabicommerce?tab=readme-ov-file#total-time):

![](https://raw.githubusercontent.com/Xcelled/mabicommerce/screenshots/dunby_waypoints.png)

If you can't get actual in-game coordinates easily, you could probably fake it by using the x/y values of pixels on the map images themselves - so pick a pixel on the minimap image and then use the X/Y of that pixel as your temporary waypoint. You'll need to change from "minimap coordinates" to "world coordinates" using a calculation similar to this one: `world_x = (minimap_x / minimap_width) * world_width`. The world region width should be in the client data files as well.

You need to manually input the connection between waypoints so that MabiCommerce knows which waypoints should be considered connected. This is how MabiCommerce knows not to try to route you through a wall, for example.

**Comment by Queenhelena** [comment]
and where exactly are the pack files located if you remember

**Comment by Xcelled** [comment]
You'll need to follow a guide on how to extract the mabi game data. Some examples from a quick search to get you started:

- https://mabi.pro/thread/151-modding-start-dungeon-defog-big-images
- https://mabimods.net/index.php?topic=14915.0
-  potentially this tool for newer .it format: https://github.com/shaggyze/mabi-pack2/releases

**Comment by nontechsavy** [comment]
Hey Queenhelena, are you willing to share the updated files at all? I've been trying to follow the instructions above with no success. Thanks!

**Comment by nontechsavy** [comment]
> im trying to update it for irias stuff but dont know where you got some of the data so maybe you be down to share your discord and we can chat some time

@Queenhelena are you willing to share the updated files at all? I've been trying to follow the instructions above with no success. Thanks!

**Comment by Queenhelena** [comment]
@nontechsavy i ran into issues with alot of the older stuff not working properly so its above my knowledge
